src/inmanta/deploy/scheduler.py

```diff
@@ -628,6 +628,18 @@ async def report_resource_state(
                             deploying=set(),
                         )
 
+    def get_last_non_deploying_state_for_dependencies(
+        self, resource_id: ResourceIdStr
+    ) -> dict[ResourceIdStr, const.ResourceState]:
+        """
+        Get resource state for every dependency of a give resource from the scheduler state
+
+        :param resource_id: The id of the resource to find the dependencies for
+        """
+        provides_view: Mapping[ResourceIdStr, Set[ResourceIdStr]] = self._state.requires
+        dependencies: Set[ResourceIdStr] = provides_view.get(resource_id, set())
+        return {id: const.ResourceState(self._state.resource_state[id].deployment_result) for id in dependencies}
+
```

Comment:
the `const.ResourceState` does not match 1-to-1 to the `ResourceState` in `self._state.resource_state[id]`. The only states that are accounted for are `deployed` and `failed` (There is also the `new` state which was not defined on `const.ResourceState` and if it happens here it should crash).

How should this be handled? Do we only need to take into account the `deployed` and `failed` states?

Reply:
you need to convert the state from one form to the other. 

(the table is already in some testcase, but it may need some TLC)